File: app/Fantasy/backend/Models/FantasyLeague.cs
using System.ComponentModel.DataAnnotations;

namespace FantasyApi.Models;

public class FantasyLeague
{
    [Key]
    public Guid Id { get; set; } = Guid.NewGuid();
    public Person? leagueAdmin { get; set; }
    public string? leagueAdminId { get; set; }
    public List<FantasyTeam>? fantasyTeams { get; set; }
    public required string LeagueName { get; set; }
    public string? JoinCode { get; set; }
}

File: app/Fantasy/frontend/src/models/FantasyLeague.ts
import type { User } from "./User";

export type FantasyLeague = {
  id: string;
  leagueAdmin: User;
  leagueAdminId: string;
  fantasyTeams: string[];
  leagueName: string;
  joinCode: string;
  teamName: string;
  teamId: string;
  isDraftStarted: boolean;
};
